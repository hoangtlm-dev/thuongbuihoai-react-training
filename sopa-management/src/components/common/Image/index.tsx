// Libs
import React, { memo } from 'react';
import { Link } from 'react-router-dom';

interface ImageProps {
  url?: string;
  alt?: string;
  href?: string;
  width: number;
  height: number;
  className?: string;
}

const Image: React.FC<ImageProps> = ({
  url = '',
  alt = '',
  href,
  height,
  width,
  className = ''
}) =>
  href ? (
    <Link data-testid='image-link' to={href} aria-label='link to home page'>
      <img className={className} src={url} alt={alt} width={width} height={height} />
    </Link>
  ) : (
    <img data-testid='image' className={className} src={url} alt={alt} />
  );

export default memo(Image);
