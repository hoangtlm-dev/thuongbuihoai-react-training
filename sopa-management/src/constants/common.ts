const SIZE = [
  {
    key: "1",
    label: "Small",
  },
  {
    key: "2",
    label: "Medium",
  },
  {
    key: "3",
    label: "Large",
  }
];

const USER = [
  {
    key: "1",
    label: "Logout"
  },
];

export {
  SIZE,
  USER,
};
