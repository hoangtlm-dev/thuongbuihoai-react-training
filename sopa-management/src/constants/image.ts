export const IMAGE = {
  whiteLogo: "/images/logos/white-logo.svg",
  blackLogo: "/images/logos/black-logo.svg",
  atl: "This is logo sopa",
};
