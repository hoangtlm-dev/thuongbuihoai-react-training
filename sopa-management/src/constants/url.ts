const BASE_URL = process.env.VITE_BASE_URL;
const PRODUCT_URL = `${BASE_URL}/products`;
const ACCOUNT_URL = `${BASE_URL}/accounts`;

export {
  PRODUCT_URL,
  ACCOUNT_URL
};
