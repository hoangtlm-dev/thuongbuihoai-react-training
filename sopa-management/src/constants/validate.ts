const CONFIRM_MESSAGE = {
  ADD_SUCCESS: 'Add product to cart successfully',
  DELETE_SUCCESS: 'Delete product to cart successfully',
  CONFIRM_DELETE: 'Are you sure you want to delete?'
}

const ERROR_MESSAGES = {
  REQUIRED: 'This field is required',
  EMAIL_FORMAT: 'Wrong email format',
  EMAIL_INCORRECT: 'Email is incorrect',
  PASSWORD_FORMAT: 'Password must be more than 6 characters',
  PASSWORD_INCORRECT: 'Password is incorrect',
}

export {
  CONFIRM_MESSAGE,
  ERROR_MESSAGES,
};
