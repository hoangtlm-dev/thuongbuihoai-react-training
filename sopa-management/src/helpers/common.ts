// Flattens an array of arrays into a single array
export const flattenArray = <T>(pages: T[][]): T[] => {
  const result: T[] = [];
  pages.forEach((page) => {
    result.push(...page);
  });

  return result;
};
