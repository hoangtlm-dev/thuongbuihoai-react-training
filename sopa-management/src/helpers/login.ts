import {
  Account,
  LoginType
} from "@interfaces/account";
import { FieldError } from "react-hook-form";

export enum CheckType {
  login = 'login',
  email = 'email',
  password = 'password'
}

// Check login
export const checkAccount = (
  data?: Account[],
  email?: string,
  password?: string,
  checkType?: CheckType,
) => {
  switch (checkType) {
    case CheckType.login:
      return data?.some((user: Account) => user.email === email && user.password === password);
    case CheckType.email:
      return data?.some((user: Account) => user.email === email);
    case CheckType.password:
      return data?.some((user: Account) => user.password === password);
    default:
      throw new Error('Invalid check type');
  }
};

export const checkValidationStyles = (
  serverError: boolean,
  rules: FieldError | undefined,
  isDirty: boolean,
  type: LoginType,
) => {
  const checkIsDirty = isDirty ? type.info : type.default
  const style = serverError || rules ? type.error : checkIsDirty;

  return { style };
};