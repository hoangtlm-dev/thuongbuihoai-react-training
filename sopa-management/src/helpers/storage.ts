import { Product } from "../interfaces/product";

export const setStorage = (key: string, data: Product) => {
  localStorage.setItem(key, JSON.stringify(data));
};

export const getStorage = (key: string) => {
  return JSON.parse(localStorage.getItem(key) as any) || [];
};

export const clearStorage = () => {
  localStorage.clear();
};
