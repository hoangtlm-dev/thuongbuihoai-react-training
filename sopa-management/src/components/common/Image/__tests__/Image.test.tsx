// Libs
import '@testing-library/jest-dom';

// Constants
import { IMAGE } from '../../../../constants/image';

// Helpers
import { renderRouterTest } from '../../../../helpers/testUtils';

// Components
import Image from '..';

describe('Image component', () => {
  test('should render Image component', () => {
    const { getByTestId } = renderRouterTest(
      <Image url={IMAGE.blackLogo} href='/' />
    );
    expect(getByTestId('image-link')).toBeInTheDocument();
  });

  test('should render Image component default url', () => {
    const { getByTestId } = renderRouterTest(
      <Image url={IMAGE.blackLogo} href='/' />
    );
    expect(getByTestId('image-link')).toBeInTheDocument();
  });

  test('should render Image component default', () => {
    const { getByTestId } = renderRouterTest(
      <Image />
    );
    expect(getByTestId('image')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = renderRouterTest(
      <Image url={IMAGE.blackLogo} href='/' />
    );
    expect(container).toMatchSnapshot();
  });
});
