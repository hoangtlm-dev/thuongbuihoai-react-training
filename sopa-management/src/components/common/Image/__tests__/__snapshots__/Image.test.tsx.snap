// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component matches snapshot 1`] = `
<DocumentFragment>
  <a
    data-testid="image-link"
    href="/"
  >
    <img
      alt=""
      class=""
      src="/images/logos/black-logo.svg"
    />
  </a>
</DocumentFragment>
`;
